File: Multilanguage_puzzles/robot-show/robot_show.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

int main()
{
    int L;
    scanf("%d", &L);
    int N;
    scanf("%d", &N);
    int left, right;
    left = L;
    right = 0;
    for (int i = 0; i < N; i++) {
        int b;
        scanf("%d", &b);
        fprintf(stderr, "%d ", b );
        if (b < left)
        {
            left = b;
        }
        if (b > right){
            right = b;
        }
    }
    fprintf(stderr, "\nmin=%d max=%d\n", left, right);
    if ( left < L - right)
    {
        printf("%d\n", L - left);
    }
    else
    {
         printf("%d\n", L - L + right);
    }
    return 0;
}

File: Multilanguage_puzzles/robot-show/robot_show.php
<?php
fscanf(STDIN, "%d", $L);
fscanf(STDIN, "%d", $N);
$inputs = explode(" ", fgets(STDIN));
$arr = array();
for ($i = 0; $i < $N; $i++)
{
    $b = intval($inputs[$i]);
    $arr[] = $b;
}
error_log(var_export($arr, true));
$left = min($arr);
$right = max($arr);
$time = $L - min($left, $L - $right);
echo((string)$time);
?>